packages/xml-views-completion/src/providers/attributeValue/boolean-literal.ts

```diff
@@ -0,0 +1,54 @@
+import { map, find } from "lodash";
+import { XMLAttribute } from "@xml-tools/ast";
+import { flattenProperties } from "@ui5-language-assistant/logic-utils";
+import {
+  BooleanValueInXMLAttributeValueCompletion,
+  BooleanValue,
+} from "../../../api";
+import {
+  getClassByElement,
+  filterMembersForSuggestion,
+} from "../utils/filter-members";
+import { UI5AttributeValueCompletionOptions } from "./index";
+
+/**
+ * Suggests boolean value (true/false) inside attribute value
+ */
+export function booleanSuggestions(
+  opts: UI5AttributeValueCompletionOptions
+): BooleanValueInXMLAttributeValueCompletion[] {
+  const xmlElement = opts.element;
+  const xmlAttribute = opts.attribute;
+
+  const elementClass = getClassByElement(xmlElement, opts.context);
+  if (elementClass === undefined) {
+    return [];
+  }
+  const properties = flattenProperties(elementClass);
+  const ui5Property = find(properties, ["name", xmlAttribute.key]);
+  const propType = ui5Property?.type;
+  if (propType?.kind !== "PrimitiveType" || propType.name !== "Boolean") {
+    return [];
+  }
+
+  const values: BooleanValue[] = [
+    {
+      kind: "BooleanValue",
+      name: "true",
+      value: true,
+    },
+    {
+      kind: "BooleanValue",
+      name: "false",
+      value: false,
+    },
+  ];
+  const prefix = opts.prefix ?? "";
+  const prefixMatchingValues = filterMembersForSuggestion(values, prefix, []);
+
+  return map(prefixMatchingValues, (_) => ({
```

Comment:
subjective:

From debugging perspective it may be best to assign this result to a variable
and then return it in a separate statement.

Reply:
Done